apps/src/code-studio/initApp/project.js

```diff
@@ -868,6 +871,22 @@ var projects = module.exports = {
     }
   },
 
+  saveSelectedSong(id) {
+    return new Promise(resolve => {
+      this.getUpdatedSourceAndHtml_(sourceAndHtml => {
+        this.saveSourceAndHtml_(
+          {
+            ...sourceAndHtml,
+            selectedSong: id,
+          },
+          () => {
+            resolve();
+          }
```

Comment:
Could replace
```js
() => {
  resolve();
}
```
with
```js
resolve
```
unless you're trying to ensure we throw away any arguments to the callback.